test/meson.build

```diff
@@ -0,0 +1,33 @@
+check = dependency('check', required: false)
+deps = [udev, glib2, check, libsystemd, libmiracle_shared_dep]
+
+if check.found()
+  test_rtsp = executable('test_rtsp', 'test_rtsp.c', dependencies: deps)
+
+  test_wpas = executable('test_wpas', 'test_wpas.c', dependencies: deps)
+
+  test_valgrind = executable('test_valgrind',
+    'test_valgrind.c',
+    dependencies: deps
+  )
+
+  valgrind = find_program('valgrind')
+
+#  set(VALGRIND CK_FORK=no valgrind --tool=memcheck --leak-check=yes --show-reachable=yes --leak-resolution=high --error-exitcode=1 --suppressions=${CMAKE_SOURCE_DIR}/test.supp)
```

Comment:
Valgrind support is already built into the meson test suite, so you should just be passing the executables to `test()` you can then run `mesontest --wrapper=valgrind $builddir`